src/routes/migration/container/index.tsx

```diff
@@ -0,0 +1,27 @@
+import React, { useEffect } from 'react'
+
+export type MigrationMessage = {
+  migrate: boolean
+  payload: string
+}
+
+const MigrationScreen: React.FC = () => {
+  useEffect(() => {
+    const loadStorageMigrationDone = async () => {
+      const payload = {}
+      Object.keys(localStorage).forEach((key) => {
+        payload[key] = JSON.stringify(localStorage[key])
+      })
+      const message: MigrationMessage = {
+        migrate: true,
+        payload: JSON.stringify(payload),
+      }
+      window.parent.postMessage(message, '*')
```

Comment:
`*` will expose your private localStorage to any website that includes the safe as an iframe.

Reply:
You're right, but if i don't put `*` i can't post to the parent since they don't share same url or port. There's two options, i can create an if clause to check that the parent window is the one that we want to migrate to, or in stead of migrating to the parent window, create an iframe(loading the migration target) and post to that url

Reply:
The parent window will always have one and the same URL (per environment).